gtsam/hybrid/tests/testIncrementalHybrid.cpp

```diff
@@ -45,296 +44,334 @@ using symbol_shorthand::X;
-// }
+/* ****************************************************************************/
+// Test if we can incrementally do the inference
+TEST(DCGaussianElimination, Incremental_inference) {
+  Switching switching(3);
+  IncrementalHybrid incrementalHybrid;
+  GaussianHybridFactorGraph graph1;
+
+  // Create initial factor graph
+  // *- X1 -*-  X2 -*-X3
+  //     \*-M1-*/
+  graph1.push_back(switching.linearizedFactorGraph.dcGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(1));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(2));
+
+  // Create ordering.
+  Ordering ordering;
+  ordering += X(1);
+  ordering += X(2);
 
-/* ************************************************************************* */
-/* Test for figuring out the optimal ordering to ensure we get a discrete graph
- * after elimination. */
+  // Run update step
+  incrementalHybrid.update(graph1, ordering);
+
+  // Test if hybrid bayes net is the correct size and if the keys are as
+  // expected.
+  auto hybridBayesNet = incrementalHybrid.hybridBayesNet();
+  EXPECT_LONGS_EQUAL(2, hybridBayesNet.size());
+  EXPECT(hybridBayesNet.at(0)->frontals() == KeyVector{X(1)});
+  EXPECT(hybridBayesNet.at(0)->parents() == KeyVector({X(2), M(1)}));
+  EXPECT(hybridBayesNet.at(1)->frontals() == KeyVector{X(2)});
+  EXPECT(hybridBayesNet.at(1)->parents() == KeyVector({M(1)}));
+
+  // Check if the remaining factor graph and DecisionTreeFactor of discrete
+  // modes is the right size
+  auto remainingFactorGraph = incrementalHybrid.remainingFactorGraph();
+  EXPECT_LONGS_EQUAL(1, remainingFactorGraph.size());
+
+  auto discreteFactor_m1 = *dynamic_pointer_cast<DecisionTreeFactor>(
+      remainingFactorGraph.discreteGraph().at(0));
+  EXPECT(discreteFactor_m1.keys() == KeyVector({M(1)}));
+
+  /********************************************************/
+  // New factor graph for incremental update.
+  GaussianHybridFactorGraph graph2;
+
+  graph2.push_back(
+      switching.linearizedFactorGraph.dcGraph().at(1));  // p(x3 | x2, m2)
+  graph2.push_back(switching.linearizedFactorGraph.gaussianGraph().at(3));
+
+  // Create ordering.
+  Ordering ordering2;
+  ordering2 += X(2);
+  ordering2 += X(3);
+
+  incrementalHybrid.update(graph2, ordering2);
+
+  // Test if hybrid bayes net is the correct size and if the keys are as
+  // expected.
+  auto hybridBayesNet2 = incrementalHybrid.hybridBayesNet();
+  EXPECT_LONGS_EQUAL(3, hybridBayesNet2.size());
+  EXPECT(hybridBayesNet2.at(1)->frontals() == KeyVector{X(2)});
+  EXPECT(hybridBayesNet2.at(1)->parents() == KeyVector({X(3), M(2), M(1)}));
+  EXPECT(hybridBayesNet2.at(2)->frontals() == KeyVector{X(3)});
+  EXPECT(hybridBayesNet2.at(2)->parents() == KeyVector({M(2), M(1)}));
+
+  // Check if the remaining factor graph and DecisionTreeFactor of discrete
```

Comment:
// Check that the remaining factor graph has a single decision tree factor on M2 and M1.